### ISSUE TITLE
Test suite very faily

### ISSUE THREAD

**Issue by daleharvey** [pouchdb/pouchdb#6831]
It looks like firefox tests are extremely flaky, and is hardcoded to 47 which is pretty old

**Comment by dharders** [comment]
I think I found the culprit test causing the 'flakiness' @daleharvey

I believe it was `test.attachments.js  #3961 Many attachments on same doc`. I was able to replicate the constant failures with CouchDB 1.6.1 and Firefox 47 (and latest 56.0.2, so I think hardcoding can remain for now) on my local dev setup.

Lowering the attachments on the same doc from 100 to 30 seemed to fix the random timeouts. I found that this particular combination of firefox + couch 1.6 + the number of attachments + the preceding tests before it (i.e. local:http, http:http, then http:local) it would max out connections and return 404s hence fail.

If you GREP the 3961 in isolation it wouldn't happen, but if you GREP=test.attachments and let it run through all adapter combos I could reproduce it every time on my machine.

I've been hitting this the last couple of days with PR #6230 and think I've nailed it so added a commit there [see test.attachments.js](https://github.com/pouchdb/pouchdb/pull/6833/commits/d31871117a38707683350877c655adc8d7dd9098#diff-d1b96de167a5748316da6556fd384221) ( to see if it fixes selenium:firefox failures. Travis is still running so I'll report back once complete)!.

Lowering the number of attachments doesn't seem to alter the assertion of the test either, as it only checks that all attachments survive the replication etc. The firefox failures come from timeout random weirdness.

**Comment by daleharvey** [comment]
Awesome job, so that test is there specifically to diagnose a large number of attachments and saturated connections so I dont think reducing the limit is a good idea. However I do think it would be fine to unblock other development to open a bug for that test, skip it and reference the open bug in comments next to the test to say why its skipped

**Comment by dharders** [comment]
Done.

1. Tracking flaky test in new issue #6835 
2. Reverted limit reduction back to original value of 100. [commit 0ac9ba5](https://github.com/pouchdb/pouchdb/pull/6833/commits/0ac9ba58b181c66240b784fda95b19309f595e94)
3. Skipped test and made a comment why with a link to this issue and the new tracking issue. [commit cc40d38](https://github.com/dharders/pouchdb/commit/cc40d3893aeaa3943680b7ab054a4eccd718ecee)

**Comment by daleharvey** [comment]
Thats great, thanks so much, any chance you can make a PR out of that?

**Comment by dharders** [comment]
@daleharvey I've got it in PR #6833 at the moment.

Do you want me to extract it into its own PR and remove it from #6833?

**Comment by dharders** [comment]
@daleharvey 

PS is travis paused at the minute ? Nothing is running... just pending

![travis](https://user-images.githubusercontent.com/16676192/32694318-0bc8850a-c791-11e7-989a-60f52adbbc8d.png)

**Comment by dharders** [comment]
NVM, they were just running one at a time sequentially (I thought they always batched a few in parallel 😕 )

**Comment by daleharvey** [comment]
@dharders Hey, yeh if you could split them out it would be great, sometimes for small issues its fine to put them in the same PR, but especially for test stuff it can be inconvenient if anything needs backed out or just to see when something fixed (or whether its actually broken)

**Comment by dharders** [comment]
Created PR #6842

**Comment by daleharvey** [comment]
That pr landed and the test suite is vastly greener :)
